src/test/java/sdmExam/BoardShould.java

```diff
@@ -1,75 +1,113 @@
         board.addStoneAt(Stone.WHITE, Position.in(12, 5));
         board.addStoneAt(Stone.WHITE, Position.in(13, 5));
-        Intersection intersection = board.intersectionAt(Position.in(12, 5));
-        assertTrue(board.existsOrthogonallyAdjacentWithStone(intersection, Stone.WHITE));
-    }
 
-    @Test
-    public void noDiagonalAdjacencyOfIntersection() {
-        board.addStoneAt(Stone.WHITE, Position.in(7, 9));
-        Intersection intersection = board.intersectionAt(Position.in(7, 9));
-        assertFalse(board.existsDiagonallyAdjacentWithStone(intersection, Stone.WHITE));
+        Intersection firstIntersection = board.intersectionAt(Position.in(7, 9));
+        Intersection secondIntersection = board.intersectionAt(Position.in(3, 4));
+        Intersection thirdIntersection = board.intersectionAt(Position.in(12, 5));
+
+        List<Intersection> inputList = Arrays.asList(firstIntersection, secondIntersection, thirdIntersection);
+        List<Boolean> outputList = Arrays.asList(false, true, true);
+
+        return inputList.stream()
+                .map(intersection -> DynamicTest.dynamicTest("Checking Orthogonal Adjacent of "+ intersection,
+                        () -> {
+                            int id = inputList.indexOf(intersection);
+                            assertEquals(outputList.get(id),board.existsOrthogonallyAdjacentWithStone(intersection,Stone.WHITE));
+                        }));
     }
 
-    @Test
-    public void upRightDiagonalAdjacencyOfIntersection() {
+
+    @TestFactory
+    Stream<DynamicTest> checkDiagonalAdjacent() {
+        board.addStoneAt(Stone.WHITE, Position.in(7, 9));
         board.addStoneAt(Stone.WHITE, Position.in(3, 4));
         board.addStoneAt(Stone.WHITE, Position.in(2, 5));
-        Intersection intersection = board.intersectionAt(Position.in(3, 4));
-        assertTrue(board.existsDiagonallyAdjacentWithStone(intersection, Stone.WHITE));
-    }
-
-    @Test
-    public void downLeftDiagonalAdjacencyOfIntersection() {
         board.addStoneAt(Stone.WHITE, Position.in(12, 5));
         board.addStoneAt(Stone.WHITE, Position.in(13, 4));
-        Intersection intersection = board.intersectionAt(Position.in(12, 5));
-        assertTrue(board.existsDiagonallyAdjacentWithStone(intersection, Stone.WHITE));
+
+        Intersection firstIntersection = board.intersectionAt(Position.in(7, 9));
+        Intersection secondIntersection = board.intersectionAt(Position.in(3, 4));
+        Intersection thirdIntersection = board.intersectionAt(Position.in(12, 5));
+
+        List<Intersection> inputList = Arrays.asList(firstIntersection, secondIntersection, thirdIntersection);
+        List<Boolean> outputList = Arrays.asList(false, true, true);
+
+        return inputList.stream()
+                .map(intersection -> DynamicTest.dynamicTest("Checking Diagonal Adjacent of "+ intersection,
+                        () -> {
+                        int id = inputList.indexOf(intersection);
```

Comment:
Rename id to index, it is more a suitable name

Reply:
Done